zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +133,137 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() { // Test multiple small transactions to fill a block max size
+    let blockheader = generate_block_header();
+
+    // 23511 transactions = data.len() = 1999925 bytes
+    // 23512 transactions = data.len() = 2000010 bytes
+    let mut limit_txs = 23511;
```

Comment:
Suggestion: Can you calculate this number based on the maximum block size `MAX_BLOCK_BYTES` ?

If you use @yaahc's suggestion of collect and repeat, you can use a constant:
```suggestion
    let MAX_TRANSACTIONS_IN_BLOCK = MAX_BLOCK_BYTES / zebra_test::vectors::DUMMY_TX1.len();
```

Reply:
Oh, but you would also need to use the header size.

Maybe don't worry about it, if the calculation is too complex.

Reply:
Applied at https://github.com/ZcashFoundation/zebra/pull/558/commits/58c11c33affafbfecd89e49c940ee8dc3082d4cb